sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BeamBigQuerySqlDialect.java

```diff
@@ -157,8 +166,13 @@ public void unparseCall(
         break;
       case OTHER_FUNCTION:
         String funName = call.getOperator().getName();
-        if (NUMERIC_LITERAL_FUNCTION.equals(funName)) {
-          // self-designed function dealing with the unparsing of ZetaSQL numeric literal
+        if (DOUBLE_FUNCTIONS.containsKey(funName)) {
+          // self-designed function dealing with the unparsing of ZetaSQL DOUBLE positive
+          // infinity, negative infinity and NaN
+          unparseDoubleWrapperFunction(writer, call);
```

Comment:
Actually you can just pass `funName` to it.

Reply:
And after you change this, could you please do a rebase again?

Reply:
Acknowledged.